src/mfa.fish

```diff
@@ -0,0 +1,63 @@
+#!/usr/bin/env fish
+
+# you MUST source this file to get the exports in your shell!
+if test "$_" = source || test -n "$IS_SOURCED"
```

Comment:
`IS_SOURCED` is a hacky workaround for the fact that fish doesn't set `$_` the same when it is sourced from inside a command substitution.  Probably a better way to do this.

Reply:
It doesn't set it the same, or not at all?

Reply:
That is a good question, I'll have to test that specifically.  If sourcing indirectly, like `fish -c 'source ...'` that test returns false.

Reply:
Add an echo line to see what gets set, and maybe you can still use whatever is there

Reply:
This is actually broken already, based on basic testing (it does not work for `.`, only `source`:

Script:
```
#!/usr/bin/env fish

echo "$_"
```

Output:
```
. test.sh 
.
chris.rogus@BZUSC02YT0FALVCH ~> source test.sh
source
```

Reply:
So I think this specific line should instead be something like:

```
#!/usr/bin/env fish

set IS_SOURCED_COMMANDS . source fish
echo $IS_SOURCED_COMMANDS
set SOURCE (status current-command)
echo $SOURCE
if contains $SOURCE $IS_SOURCED_COMMANDS
  set SOURCED 1
else
  set SOURCED 0
end
echo $SOURCED
```

HOWEVER... this reveals that:
```
chris.rogus@BZUSC02YT0FALVCH ~> . test.sh
. source fish
.
1
chris.rogus@BZUSC02YT0FALVCH ~> source test.sh
. source fish
source
1
chris.rogus@BZUSC02YT0FALVCH ~> fish test.sh
. source fish
fish
1
chris.rogus@BZUSC02YT0FALVCH ~> ./test.sh
. source fish
fish
1
```
^^ note that we explicitly do not want that last one to work... Or do we? So, in bash, that last case definitely means that env vars will not be set, which is why sourcing is a hard requirement. But... perhaps fish setting env vars in a script will actually work outside, regardless of source-ing the file. So, maybe this requirement is no longer applicable for fish??

Reply:
Hmm, no, as it happens, fish does the same thing as bash: you need to source the file to get env vars. UNLESS you do `set -xU` which makes it available to all users shells across the machine. But this seems like a very dangerous (security risk) move, so I would avoid it. That means we do not want to include `fish` in the list of acceptable sources, and that means we need to set an env var before running the `fish -c ...` tests. But we can make things easier by doing something simpler (most definitely cheating) like: `if contains $SOURCE $SOURCES; or $SOURCED; set $SOURCED 1; end` something like that

Reply:
In short, at this point, I think the only thing that needs to change here, for sure, is: accept `.` as well as `source`. That said, I suggest also using `status current-command` instead of `$_`. Past that, what you have here as your workaround for `fish -c ...` seems acceptable.

Reply:
Great find here, I've added what you suggested.  Also, digging how expressive it is to do things like see if a term is contained in a list.